app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.__class__.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def die(self) -> None:
+        if self in self.__class__.alive:
+            self.__class__.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(prey: Animal) -> None:
+        if not isinstance(prey, Herbivore) or prey.hidden:
+            return
```

Comment:
The condition `not isinstance(prey, Herbivore) or prey.hidden` in the `Carnivore.bite` method might not work as intended. If `prey` is not a `Herbivore`, the method will return immediately, which is correct. However, if `prey` is a `Herbivore` but `prey.hidden` is `True`, it will also return, which is correct. Ensure this logic aligns with the intended behavior: `Carnivore` should only attack visible `Herbivore` instances.